Backtracking/GenerateSubsets

```diff
@@ -0,0 +1,26 @@
+function generateSubsets(nums) {
```

Comment:
1. Call the parameter `arr`, not `nums`. The implementation doesn't need to make any assumptions about the values in `arr`.
2. This should have a doc comment stating what this does (builds and returns a list of all subsets in lexicographic order). Also, consider using a generator rather than building a list; sometimes you want to iterate over all subsets without storing them all temporarily in memory (for example when solving hard optimization problems).